From stable-bounces@linux.kernel.org  Mon Jan 16 22:19:55 2006
Message-Id: <200601170615.k0H6FDZ6028782@shell0.pdx.osdl.net>
To: torvalds@osdl.org
From: akpm@osdl.org
Date: Mon, 16 Jan 2006 22:14:55 -0800
Cc: nickpiggin@yahoo.com.au, axboe@suse.de, 76306.1226@compuserve.com, stable@kernel.org
Subject: [PATCH] elevator=as back-compatibility

From: Chuck Ebbert <76306.1226@compuserve.com>

As of 2.6.15 you need to use "anticipatory" instead of "as".  Fix that up
so that `elevator=as' still works.

Signed-off-by: Chuck Ebbert <76306.1226@compuserve.com>
Cc: Nick Piggin <nickpiggin@yahoo.com.au>
Cc: Jens Axboe <axboe@suse.de>
Cc: <stable@kernel.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Chris Wright <chrisw@sous-sol.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 block/elevator.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- linux-2.6.15.1.orig/block/elevator.c
+++ linux-2.6.15.1/block/elevator.c
@@ -150,6 +150,13 @@ static void elevator_setup_default(void)
 	if (!chosen_elevator[0])
 		strcpy(chosen_elevator, CONFIG_DEFAULT_IOSCHED);
 
+	/*
+	 * Be backwards-compatible with previous kernels, so users
+	 * won't get the wrong elevator.
+	 */
+	if (!strcmp(chosen_elevator, "as"))
+		strcpy(chosen_elevator, "anticipatory");
+
  	/*
  	 * If the given scheduler is not available, fall back to no-op.
  	 */
